automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/BatchNodepoolCreationSpec.scala

```diff
@@ -0,0 +1,171 @@
+package org.broadinstitute.dsde.workbench.leonardo
+
+import java.nio.file.Paths
+
+import cats.effect.IO
+import cats.implicits._
+import com.google.container.v1.{Cluster, NodePool}
+import org.broadinstitute.dsde.workbench.DoneCheckable
+import org.broadinstitute.dsde.workbench.auth.AuthToken
+import org.broadinstitute.dsde.workbench.google2.GKEModels.{KubernetesClusterId, KubernetesClusterName}
+import org.broadinstitute.dsde.workbench.google2.{streamFUntilDone, DiskName, GKEService}
+import org.broadinstitute.dsde.workbench.leonardo.LeonardoApiClient._
+import org.broadinstitute.dsde.workbench.leonardo.http.{GetAppResponse, ListAppResponse, PersistentDiskRequest}
+import org.http4s.headers.Authorization
+import org.http4s.{AuthScheme, Credentials}
+import org.scalatest.{DoNotDiscover, ParallelTestExecution}
+
+import scala.collection.JavaConverters._
+import scala.concurrent.duration._
+
+@DoNotDiscover
+class BatchNodepoolCreationSpec
+    extends GPAllocFixtureSpec
+    with LeonardoTestUtils
+    with GPAllocUtils
+    with ParallelTestExecution {
+
+  implicit val ronToken: AuthToken = ronAuthToken
+  implicit val auth: Authorization = Authorization(Credentials.Token(AuthScheme.Bearer, ronCreds.makeAuthToken().value))
+
+  val gkeServiceResource = GKEService.resource(Paths.get(LeonardoConfig.GCS.pathToQAJson), blocker, semaphore)
+
+  //the cluster and nodepools should be running
+  val clusterDoneCheckable: DoneCheckable[Option[Cluster]] =
+    x =>
+      x.map(_.getStatus()) == Some(Cluster.Status.RUNNING) &&
+        x.map(_.getNodePoolsList().asScala.toList.map(_.getStatus()).distinct) == Some(List(NodePool.Status.RUNNING))
+
+  "batch nodepool creation should work" - {
+    withNewProject { googleProject =>
+      val test = LeonardoApiClient.client.use { c =>
+        implicit val httpClient = c
+        for {
+          clusterName <- IO.fromEither(KubernetesNameUtils.getUniqueName(KubernetesClusterName.apply))
+          _ <- LeonardoApiClient.batchNodepoolCreate(googleProject,
+                                                     defaultBatchNodepoolRequest.copy(clusterName = Some(clusterName)))
+          getCluster = gkeServiceResource.use { gkeClient =>
+            val id = KubernetesClusterId(googleProject, LeonardoConfig.Leonardo.location, clusterName)
+            gkeClient.getCluster(id)
+          }
+          monitorCreationResult <- testTimer.sleep(30 seconds) >> streamFUntilDone(getCluster, 20, 10 seconds)(
+            testTimer,
+            clusterDoneCheckable
+          ).compile.lastOrError
+
+          _ = monitorCreationResult.map(_.getNodePoolsList().size()) shouldBe Some(
+            defaultBatchNodepoolRequest.numNodepools.value + 1
+          )
+        } yield ()
+      }
+
+      test.unsafeRunSync()
```

Comment:
Are we relying on gpalloc to delete the cluster (e.g. there is not explicit deletion in the test)?

Reply:
Yes. i could add this though, but i know @gpcarr  is working on this currently